astroid/nodes/_base_nodes.py

```diff
@@ -223,3 +250,526 @@ def _elsed_block_range(
                 return lineno, orelse[-1].tolineno
             return lineno, orelse[0].fromlineno - 1
         return lineno, last or self.tolineno
+
+
+class LookupMixIn(NodeNG):
+    """Mixin to look up a name in the right scope."""
+
+    @lru_cache  # noqa
+    def lookup(self, name: str) -> tuple[LocalsDictNodeNG, list[NodeNG]]:
+        """Lookup where the given variable is assigned.
+
+        The lookup starts from self's scope. If self is not a frame itself
+        and the name is found in the inner frame locals, statements will be
+        filtered to remove ignorable statements according to self's location.
+
+        :param name: The name of the variable to find assignments for.
+
+        :returns: The scope node and the list of assignments associated to the
+            given name according to the scope where it has been found (locals,
+            globals or builtin).
+        """
+        return self.scope().scope_lookup(self, name)
+
+    def ilookup(self, name):
+        """Lookup the inferred values of the given variable.
+
+        :param name: The variable name to find values for.
+        :type name: str
+
+        :returns: The inferred values of the statements returned from
+            :meth:`lookup`.
+        :rtype: iterable
+        """
+        frame, stmts = self.lookup(name)
+        context = InferenceContext()
+        return bases._infer_stmts(stmts, context, frame)
+
+
+def _reflected_name(name) -> str:
+    return "__r" + name[2:]
+
+
+def _augmented_name(name) -> str:
+    return "__i" + name[2:]
+
+
+BIN_OP_METHOD = {
+    "+": "__add__",
+    "-": "__sub__",
+    "/": "__truediv__",
+    "//": "__floordiv__",
+    "*": "__mul__",
+    "**": "__pow__",
+    "%": "__mod__",
+    "&": "__and__",
+    "|": "__or__",
+    "^": "__xor__",
+    "<<": "__lshift__",
+    ">>": "__rshift__",
+    "@": "__matmul__",
+}
+
+REFLECTED_BIN_OP_METHOD = {
+    key: _reflected_name(value) for (key, value) in BIN_OP_METHOD.items()
+}
+AUGMENTED_OP_METHOD = {
+    key + "=": _augmented_name(value) for (key, value) in BIN_OP_METHOD.items()
+}
+
+
+class OperatorNode(NodeNG):
+    @staticmethod
+    def _filter_operation_errors(
+        infer_callable: Callable[
+            [InferenceContext | None],
+            Generator[InferenceResult | util.BadOperationMessage, None, None],
+        ],
+        context: InferenceContext | None,
+        error: type[util.BadOperationMessage],
+    ) -> Generator[InferenceResult, None, None]:
+        for result in infer_callable(context):
+            if isinstance(result, error):
+                # For the sake of .infer(), we don't care about operation
+                # errors, which is the job of pylint. So return something
```

Comment:
```suggestion
                # errors, which is the job of a linter. So return something
```
Astroid Can be used without pylint :D

Reply:
Hah, good point! I'll catch this in the next PR.